Solutions/GoogleCloudPlatformIAM/Analytic Rules/GCPIAMDisableDataAccessLogging.yaml

```diff
@@ -0,0 +1,34 @@
+id: 2530a631-9605-404d-ae58-58ef1f91b17c
+name: GCP IAM - Disable Data Access Logging
+description: |
+  'Detects when Data Access Logging is disabled.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: GCPIAMDataConnector
+    dataTypes:
+      - GCP_IAM
+queryFrequency: 1h
+queryPeriod: 1h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - DefenseEvasion
+relevantTechniques:
+  - T1562
+query: |
+  GCP_IAM
+  | where PayloadMethodname == 'SetIamPolicy'
+  | extend s = parse_json(todynamic(PayloadRequestPolicyAuditconfigs))[0]['auditLogConfigs']
+  | where isempty(s)
+  | extend AccountCustomEntity = PayloadAuthenticationinfoPrincipalemail
+  | extend IPCustomEntity = SrcIpAddr

```

Comment:
We currently have an old and a new entity mapping method. Since we are still in the transition phase, a good idea is to include both. So, in the last line of query if we can map the entity to not only IPCustomEntity but also include AccountCustomEntity as I see the entity mapping below that would be great. We should ideally also include timestamp (This is generally TimeGenerated i.e | extend timestamp = TimeGenerated) . Thanks.